### PR TITLE
MM-62197 - verify scheduled message do not exist before adding to list

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
On every team change we were validating the existing ids of the schemsg but were not checking if the id was already in the state array, so on every team change that triggered the SCHEDULED_POSTS_RECEIVED action, was causing the reducer to duplicate the entry in the array. This PR makes sure the id of the message is not present before adding it.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-62197

#### Screenshots

https://github.com/user-attachments/assets/6f3fe12d-fe79-48da-b846-6c29a48fe569

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by harshilsharma63 — APPROVED** [review]

**Review by devinbinnie — APPROVED** [review]
Asked some questions, but otherwise LGTM

File: webapp/channels/src/packages/mattermost-redux/src/reducers/entities/scheduled_posts.ts [hunk 1]
```diff
@@ -183,9 +183,14 @@ function byChannelOrThreadId(state: ScheduledPostsState['byChannelOrThreadId'] =
                 scheduledPostsByTeamId[teamId].forEach((scheduledPost: ScheduledPost) => {
                     const id = scheduledPost.root_id || scheduledPost.channel_id;
 
+                    // Check if the entry for that channel/thread ID exists
                     if (newState[id]) {
-                        newState[id].push(scheduledPost.id);
+                        // Only add if its not already there
+                        if (!newState[id].includes(scheduledPost.id)) {
```
[the remark this answers is not in the document]
Comment (pvev) [reply webapp/channels/src/packages/mattermost-redux/src/reducers/entities/scheduled_posts.ts:189]: yes, only for preventing redux information to be in an incorrect state on team switch. The other option I considered was directly resetting the scheduled messages state slice on every team switch, but this solution is simpler.

**Comment by pvev** [comment]
@marianunez can you please do a before merge review as  a QA?

**Review by marianunez — APPROVED** [review]
Stepping in to do a QA pass as Yasser is OOO. Message count does not increase on team change and is working as expected. 

Yasser can do a post-merge validation at release time.

**Comment by marianunez** [comment]
@pvev we should cherry pick this to v10.4 so this fix can go out in the January release. cc// @amyblais

**Comment by pvev** [comment]
@amyblais can this still get an automatic cherry pick to v10.4 or need to be manual?

**Pull request merged**